INDY-2138: add a documentation for a new PBFT View Change (#1431)
hyperledger/indy-plenum: docs/source/message_requests.md

- A node requests a ConsistencyProof with median rate of ledger size on other nodes.
- With the first message request for a ConsistencyProof the node is scheduling a new message request in config.ConsistencyProofTimeout for case the answer to the first request is not received.

If the node has the quorum(n - f - 1) of LedgerStatuses or the quorum(f + 1) of ConsistencyProofs, scheduled request will be canceled.

###ViewChange messages

> Loose a NewView message from primary. It is necessary to finish a view change. A node requests NewView messages from all nodes in NEW_VIEW_TIMEOUT after a view change started. 

Reviewer: ```suggestion
Lost NewView message will prevent view change from finishing successfully. A node requests NewView messages from all nodes in NEW_VIEW_TIMEOUT after a view change started. 
```

> - If an answer is received from a primary is received the node uses them and finishes the view change. 

Reviewer: ```suggestion
- If an answer is received from expected primary node uses it and finishes the view change. 
```

> - Else it uses a quorum (f+1) of NewView responses from other nodes, finishes the view change and starts catchup.

Reviewer: I'd prefer to use "otherwise" instead of "else", but this probably is subjective...

> Loose a ViewChange message from primary. It is necessary to finish a view change. A node requests missing ViewChange messages from all nodes after receiving a NewView message. 

Reviewer: ```suggestion
Lost ViewChange message can prevent view change from finishing successfully. A node requests missing ViewChange messages from all nodes after receiving a NewView message. 
```
